src/emulator/databaseEmulator.ts

```diff
@@ -60,11 +60,11 @@ export class DatabaseEmulator implements EmulatorInstance {
     return downloadableEmulators.start(Emulators.DATABASE, this.args);
   }
 
-  async connect(): Promise<void> {
-    return;
+  connect(): Promise<void> {
+    return Promise.resolve();
   }
 
-  async stop(): Promise<void> {
+  stop(): Promise<void> {
```

Comment:
In code review I have been told (maybe by you or Yuchen) that we prefer:

```
async stop() {
  await downloadableEmulators.stop(Emulators.DATABASE);
}
```

rather than removing `async` and returning the promise ... I am cool either way

Reply:
I'm fine as long as it is consistent. In this case though, the linter config alone is not enough documentation (since either your proposed snippet or Sam's code sample passes linter) so I'd like to see some additional documentation like comments or style guides.

Reply:
I am going to add this to the contributing guide I think:

1. Prefer synchronous functions. Never wrap all synchronous work in a promise (unless absolutely required).
1. When dealing with multiple promises, use `await` (and therefore `async` in the function signature) to get promise values (rather than chaining).
  1. Throwing a `FirebaseError` is acceptable within an `async` function.
1. When error handling not using `async`/`await`, strongly prefer returning a rejected promise with a `FirebaseError` rather than throwing it.